error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/ImplicitBlockingFluxOperation.java

```diff
@@ -0,0 +1,192 @@
+      description.addFix(getGuavaFix(tree, state));
+    }
+    description.addFix(getUnmodifiableListFix(tree, state));
+
+    return description.build();
+  }
+
+  private static SuggestedFix getGuavaFix(MethodInvocationTree tree, VisitorState state) {
+    SuggestedFix.Builder imports = SuggestedFix.builder();
+    String toImmutableList =
+        SuggestedFixes.qualifyStaticImport(
+            "com.google.common.collect.ImmutableList.toImmutableList", imports, state);
+    String collector = toImmutableList + "()";
+
+    return replaceMethodInvocationWithCollect(tree, collector, imports.build(), state);
+  }
+
+  private static SuggestedFix getUnmodifiableListFix(
+      MethodInvocationTree tree, VisitorState state) {
+    SuggestedFix.Builder imports = SuggestedFix.builder();
+    String toUnmodifiableList =
+        SuggestedFixes.qualifyStaticImport(
+            "java.util.stream.Collectors.toUnmodifiableList", imports, state);
+    String collector = toUnmodifiableList + "()";
+
+    return replaceMethodInvocationWithCollect(tree, collector, imports.build(), state);
+  }
+
+  // XXX: Assumes that the generated `collect(...)` expression will evaluate to
+  // `Mono<Collection<?>>`
+  // XXX: May return `emptyFix`.
+  private static SuggestedFix replaceMethodInvocationWithCollect(
+      MethodInvocationTree tree,
+      String collectArgument,
+      SuggestedFix additionalFixes,
+      VisitorState state) {
+    String collectMethodInvocation = String.format("collect(%s)", collectArgument);
+    SuggestedFix.Builder fix = replaceMethodInvocation(tree, collectMethodInvocation, state);
+    fix.merge(additionalFixes);
+
+    TypeSymbol resultTypeSymbol =
+        Optional.ofNullable(getResultType(tree)).map(Type::asElement).orElse(null);
+
+    if (Objects.isNull(resultTypeSymbol)) {
+      return SuggestedFix.emptyFix();
+    }
+
+    if (isClassValidSubstituteFor(resultTypeSymbol, Stream.class)) {
+      fix.postfixWith(tree, ".block().stream()");
+      return fix.build();
+    }
+
+    if (isClassValidSubstituteFor(resultTypeSymbol, Iterable.class)) {
+      fix.postfixWith(tree, ".block()");
+      return fix.build();
+    }
+
+    // XXX: Expected a replacement that evaluates to a type that we don't handle. Should be
+    // impossible to get here, as we only match `toStream` and `toIterable`.
+    return SuggestedFix.emptyFix();
+  }
+
+  // XXX: Assumes that the specified tree is valid, has starting position and contains the matched
+  // method invocation.
+  // XXX: Assumes that the specified tree's end is the matched method invocation's end.
+  private static SuggestedFix.Builder replaceMethodInvocation(
+      MethodInvocationTree tree, String replacement, VisitorState state) {
+    ImmutableList<ErrorProneToken> tokens =
+        ErrorProneTokens.getTokens(SourceCode.treeToString(tree, state), state.context);
+
+    int treeStartPosition = getStartPosition(tree);
+    int methodInvocationStartPosition =
```

Comment:
Nice improvements done here! 

I _think_ there might be a solution where we don't need to use the `ErrorProneToken`s. Usually we only go that route as "last resort" or if comments are involved. Now that I think of that, that _could_ also be the case here. Maybe we should add test cases for that. See the `StringCaseLocaleUsage` test cases for some examples. However, I'm not sure yet if that is the correct way to go about this either. I need to take some more time to dive into this before being able to give a good advice. For now I need to focus on some other things so will stop here :).